swarm/pss/prox_test.go

```diff
@@ -133,99 +123,132 @@ func readSnapshot(t *testing.T, nodeCount int) simulations.Snapshot {
-		for _, nod := range sim.Net.GetNodes() {
-			po, _ := pof(msgs[i], nodeAddrs[nod.ID()], 0)
-			depth := kademlias[nod.ID()].NeighbourhoodDepth()
+		// loop through all nodes and find the required and allowed recipients of each message
+		// (for more information, please see the comment to the main test function)
+		for _, nod := range d.sim.Net.GetNodes() {
+			po, _ := pof(d.msgs[i], d.nodeAddrs[nod.ID()], 0)
+			depth := d.kademlias[nod.ID()].NeighbourhoodDepth()
 
-			// only nodes with closest IDs (wrt msg) will receive the msg
-			if po > prev {
-				prev = po
+			// only nodes with closest IDs (wrt the msg address) will be required recipients
+			if po > closestPO {
+				closestPO = po
 				targets = nil
 				targets = append(targets, nod.ID())
-			} else if po == prev {
+			} else if po == closestPO {
 				targets = append(targets, nod.ID())
 			}
 
 			if po >= depth {
-				maxMessages++
-				allowed[i] = append(allowed[i], nod.ID())
-				allowedMsgs[nod.ID()] = append(allowedMsgs[nod.ID()], uint64(i))
+				d.allowedMessages++
+				d.allowed[i] = append(d.allowed[i], nod.ID())
+				d.allowedMsgs[nod.ID()] = append(d.allowedMsgs[nod.ID()], uint64(i))
 			}
 
-			// a node with the smallest PO (wrt msg) will be the sender
+			// a node with the smallest PO (wrt msg) will be the sender,
+			// in order to increase the distance the msg must travel
 			if po < smallestPo {
 				smallestPo = po
-				senders[i] = nod.ID()
+				d.senders[i] = nod.ID()
 			}
 		}
 
-		msgsToReceive += len(targets)
+		d.requiredMessages += len(targets)
 		for _, id := range targets {
-			recipients[i] = append(recipients[i], id)
-			expectedMsgs[id] = append(expectedMsgs[id], uint64(i))
+			d.recipients[i] = append(d.recipients[i], id)
+			d.expectedMsgs[id] = append(d.expectedMsgs[id], uint64(i))
 		}
 
-		log.Debug("nn for msg", "targets", len(recipients[i]), "msgidx", i, "msg", common.Bytes2Hex(msgAddr[:8]), "sender", senders[i], "senderpo", smallestPo)
+		log.Debug("nn for msg", "targets", len(d.recipients[i]), "msgidx", i, "msg", common.Bytes2Hex(msgAddr[:8]), "sender", d.senders[i], "senderpo", smallestPo)
 	}
-	log.Debug("msgs to receive", "count", msgsToReceive)
+	log.Debug("msgs to receive", "count", d.requiredMessages)
 }
 
 func TestProxNetwork(t *testing.T) {
-	if (*runNodes > 0 && *runMessages == 0) || (*runMessages > 0 && *runNodes == 0) {
-		t.Fatal("cannot specify only one of flags --nodes and --messages")
-	} else if *runNodes > 0 {
-		t.Run(fmt.Sprintf("%d/%d", *runMessages, *runNodes), testProxNetwork)
-	} else {
-		t.Run("1/4", testProxNetwork)
+	t.Run("16/16", testProxNetwork)
+}
+
+// params in run name: nodes/msgs
+func TestProxNetworkLong(t *testing.T) {
+	if !*longrunning {
+		t.Skip("run with --longrunning flag to run extensive network tests")
 	}
+	t.Run("8/100", testProxNetwork)
+	t.Run("16/100", testProxNetwork)
+	t.Run("32/100", testProxNetwork)
+	t.Run("64/100", testProxNetwork)
```

Comment:
Perhaps we should be brave and try 128 nodes, too.

Should we also try larger message counts?